src/main.ts

```diff
@@ -1,409 +1,463 @@
+      const teams: Set<string> = new Set()
+      const users: Set<string> = new Set()
+      for (const [user, { team }] of usersToAskForReview) {
+        if (team === null) {
+          users.add(user)
         } else {
-          approved_users.delete(review.user!.login)
-          console.log(`${review.state} - ${review.user!.login}`) //DEBUG
+          teams.add(team)
         }
       }
-      console.log(`Approved users: ${Array.from(approved_users)}`) //DEBUG
-
-      // check approvals
-      console.log(`###### CHECKING APPROVALS ######`) //DEBUG
-      const has_all_needed_approvals: Set<string> = new Set()
+      await octokit.request(
+        "POST /repos/{owner}/{repo}/pulls/{pull_number}/requested_reviewers",
+        {
+          owner: pr.base.repo.owner.login,
+          repo: pr.base.repo.name,
+          pull_number: pr.number,
+          reviewers: Array.from(users),
+          team_reviewers: Array.from(teams),
+        },
+      )
+    }
 
-      for (const group of final_approval_groups) {
-        const group_approvers = new Set(group.approvers)
-        const has_approvals = new Set(
-          [...group_approvers].filter((x) => approved_users.has(x)),
+    if (highestMinApprovalsRule !== null) {
+      let approvalCount = 0
+      for (const review of latestReviews.values()) {
+        if (review.isApproval) {
+          approvalCount++
+        }
+      }
+      if (approvalCount < highestMinApprovalsRule.min_approvals) {
+        problems.push(
+          `Rule ${highestMinApprovalsRule.name} requires at least ${highestMinApprovalsRule.min_approvals} approvals, but only ${approvalCount} were given`,
         )
-        console.log(
-          `Need min ${group.min_approvals} approvals from ${
-            group.approvers
-          } --- has ${has_approvals.size} - ${Array.from(has_approvals)}`,
-        ) //DEBUG
-        if (has_approvals.size >= group.min_approvals) {
-          has_all_needed_approvals.add("true")
-          pr_review_status_messages.push(
-            `${group.name} (${has_approvals.size}/${group.min_approvals})- OK!`,
-          )
+      }
+    }
+
+    if (problems.length !== 0) {
+      logger.failure("The following problems were found:")
+      for (const problem of problems) {
+        logger.log(problem)
+      }
+      logger.log("")
+      return "failure"
+    }
+  }
+
+  return "success"
+}
+
+const main = function () {
+  const context = github.context
+  if (
+    context.eventName !== "pull_request" &&
+    context.eventName !== "pull_request_review"
+  ) {
+    core.setFailed(
+      `Invalid event: ${context.eventName}. This action should be triggered on pull_request and pull_request_review`,
+    )
+    return
+  }
+
+  const logger = new Logger()
+
+  const pr = context.payload.pull_request as PR
+  const octokit = github.getOctokit(core.getInput("token", { required: true }))
+
+  const finish = async function (state: "success" | "failure") {
```

Comment:
Certainly, but I'm using those values in the Github API call payload which does not accept enums (it's a JSON payload). Rather than having to convert from enum to string, it'd rather use the strings directly which is less noisy and looks simpler.

Reply:
Maybe in this case make them constants? Like `SUCCESS` and `FAILURE`. Less likely to have a typo

Reply:
https://github.com/paritytech/pr-custom-review/pull/27/commits/048dde75ddf5ad899eb5f51e7dbfe97c113ae5b1

Reply:
Thanks :+1: